### ISSUE TITLE
GM.info / GM_info not working without @grant

### ISSUE THREAD

**Issue by wolfissimo** [quoid/userscripts#692]
According to documentation, GM.info / GM_info should be available without having to @grant it. In version 1.5.0/4.5.0 this is no longer the case, which is bad because using @grant leads to other issues. Please fix this.

**Comment by scholtzm** [comment]
Just ran into this sudden breaking change.

What's the recommended way to access GM_info now in scripts that run via `@inject-into page`?

**Comment by wolfissimo** [comment]
> So, it fits the documentation's description that you can use `GM.info` / `GM_info` without `@grant` them in the content script context.

I think, documentation is not absolutely consistent in this point:
- in "Metadata" it says "GM apis are only available when using content"
- in "API" it says "GM.info && GM_info ... is available without needing to add it to @grant"
The latter was the case until the recent update.

> For reference, could you describe your use cases in detail and the reasons you think it's required to use them in the page context?

1. My script uses a XHR proxy, which only works in page context. As soon as I add a @grant, it stops working.
2. The page my script is for is updated every few days, so I also need to update may script very often. So users should always see which script version they are running, for which I use GM.info / GM_info. So at least GM.info.script.version and GM.info.script.name should be available without @grant.

**Comment by scholtzm** [comment]
My use case is simple, I display script version from GM_info in the UI elements added by the script.

**Comment by scholtzm** [comment]
It that's what it takes, I will remove the usage of GM_info.

> I am sorry you guys are having issues for this, but as I said before, I don't think this is a "breaking change", it's more like fixing a previous accidental leak.

Agree to disagree here. Just using the API as documented in the README. This is a breaking change and the API should have been properly deprecated for use in `page` context.

Keep in mind this change might have broken hundreds of scripts that worked previously.

**Comment by wolfissimo** [comment]
> If you mean that you show the `version` of the script on the page by reading `GM.info`, I think you can do that by updating the variables in the code at the same time as you update the script, reading through `GM.info` isn't necessary is it?

Yes, thats my use case (I show version and also the script's name). But updating those values manually can easily be forgotten, and I think that is the main reason why GM.info.script.version and GM.info.script.name exist.

**Comment by wolfissimo** [comment]
> It that's what it takes, I will remove the usage of GM_info.
> 
> >

Well, I inserted the following into my script, so that users of other browsers/operating systems still enjoy the benefits of GM_info:
```
if (typeof GM_info !== 'undefined') {
    // code using GM_info ...
}
```

**Comment by scholtzm** [comment]
@wolfissimo I'm using webpack to build the script so I just adjusted the config to use DefinePlugin and replaced all GM_info references. 👍
